dbr quant: clean up AutoQuantizationState.get_op_convert_info flag

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #68432 dbr quant overhead[9/x]: precalculate when to skip op_convert_after_hook
* #68431 dbr quant: remove more asserts from hot paths
* #68404 speed up quantized relu6 inplace kernel
* #68374 dbr quant: simplify relatedness logic
* #68373 dbr quant overhead[8/x]: small speedup in op_needs_quantization
* #68372 dbr quant overhead[7/x]: speed up AutoQuantizationState.reset_to_new_call
* #68371 dbr quant overhead[6/x]: remove unneeded isinstance checks in `op_convert_before_hook`
* #68370 dbr quant overhead[5/x]: remove unnecessary asserts
* #68369 dbr quant: clean up key types in AutoQuantizationState mappings
* #68351 dbr quant overhead[4/x]: speed up hook type calculations
* #68350 dbr quant overhead[3/x]: speed up AutoQuantizationState.mark_cur_op_complete
* #68347 dbr quant overhead[2/x]: precalculate op_convert_info
* #68346 dbr quant: move model level utils into own file
* **#68345 dbr quant: clean up AutoQuantizationState.get_op_convert_info flag**
* #68344 dbr quant: refactor AutoQuantizationState._get_packed_param_name
* #68343 dbr quant: refactor AutoQuantizationState._get_input_args_quant_dequant_info
* #68342 dbr quant: refactor `get_quantized_op` to only use `seen_op_info`
* #68341 dbr quant: refactor `get_func_output_obs_type` to only use `seen_op_info`
* #68309 dbr quant overhead[1/x]: remove expensive calls to named_modules
* #68251 dbr quantization: inline scale and zp
* #67776 barebones numeric suite for define-by-run quantization

Summary:

Removes a flag to unwrap scale and zp which was only needed by
the FX rewriter. Moves the logic to happen in the FX tracer instead.
This resolves a technical debt TODO.

Test plan:

```
python test/test_quantization.py TestQuantizeDBR
```

Differential Revision: [D32463764](https://our.internmc.facebook.com/intern/diff/D32463764)

vkuzo committed:
dbr quant: clean up AutoQuantizationState.get_op_convert_info flag

Summary:

Removes a flag to unwrap scale and zp which was only needed by
the FX rewriter. Moves the logic to happen in the FX tracer instead.
This resolves a technical debt TODO.

Test plan:

```
python test/test_quantization.py TestQuantizeDBR
```

[ghstack-poisoned]

vkuzo committed:
Update on "dbr quant: clean up AutoQuantizationState.get_op_convert_info flag"


Summary:

Removes a flag to unwrap scale and zp which was only needed by
the FX rewriter. Moves the logic to happen in the FX tracer instead.
This resolves a technical debt TODO.

Test plan:

```
python test/test_quantization.py TestQuantizeDBR
```

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32463764).

jerryzh168 commented on torch/ao/quantization/_dbr/quantization_state.py:
```diff
@@ -470,12 +469,7 @@ def get_op_convert_info(
-            else:
-                additional_kwargs.update(
-                    {'scale': scale.item(), 'zero_point': zp.item()})  # type: ignore[dict-item]
+            additional_kwargs.update({'scale': scale, 'zero_point': zp})
```
why not just always call `.item` here?

vkuzo replied:
doing that leads to a runtime error, I did not debug it that closely yet.  Can follow-up if needed.

vkuzo committed:
Update on "dbr quant: clean up AutoQuantizationState.get_op_convert_info flag"


Summary:

Removes a flag to unwrap scale and zp which was only needed by
the FX rewriter. Moves the logic to happen in the FX tracer instead.
This resolves a technical debt TODO.

Test plan:

```
python test/test_quantization.py TestQuantizeDBR
```

Differential Revision: [D32463764](https://our.internmc.facebook.com/intern/diff/D32463764)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32463764).

vkuzo committed:
Update on "dbr quant: clean up AutoQuantizationState.get_op_convert_info flag"


Summary:

Removes a flag to unwrap scale and zp which was only needed by
the FX rewriter. Moves the logic to happen in the FX tracer instead.
This resolves a technical debt TODO.

Test plan:

```
python test/test_quantization.py TestQuantizeDBR
```

Differential Revision: [D32463764](https://our.internmc.facebook.com/intern/diff/D32463764)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32463764).

vkuzo committed:
Update on "dbr quant: clean up AutoQuantizationState.get_op_convert_info flag"


Summary:

Removes a flag to unwrap scale and zp which was only needed by
the FX rewriter. Moves the logic to happen in the FX tracer instead.
This resolves a technical debt TODO.

Test plan:

```
python test/test_quantization.py TestQuantizeDBR
```

Differential Revision: [D32463764](https://our.internmc.facebook.com/intern/diff/D32463764)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32463764).

vkuzo committed:
Update on "dbr quant: clean up AutoQuantizationState.get_op_convert_info flag"


Summary:

Removes a flag to unwrap scale and zp which was only needed by
the FX rewriter. Moves the logic to happen in the FX tracer instead.
This resolves a technical debt TODO.

Test plan:

```
python test/test_quantization.py TestQuantizeDBR
```

Differential Revision: [D32463764](https://our.internmc.facebook.com/intern/diff/D32463764)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32463764).
